### PR TITLE
Create an ADR for deciding whether or not to add PromptFlow.

### PR DESCRIPTION
**Pull request by superhindupur**
## Purpose
Creates an ADR for deciding whether to include promptflow to this repo. 

## Does this introduce a breaking change?

```
[ ] Yes
[x ] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test
N/A

### CODE REVIEW

File: docs/design/adrs/2024-03-18-use-cypres-as-a-framework-for-e2e-testing.md
Comment (superhindupur) [inline docs/design/adrs/2024-03-18-use-cypres-as-a-framework-for-e2e-testing.md:4]: Sorry for the many changes in this file, they're all whitespace changes apart from marking the ADR as approved

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (adamdougal) [inline docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:1]: The title of the document suggests you think we **do** want to implement this. 
```suggestion
# Do not implement a PromptFlow Orchestration as an alternative for the chat backend
```
Would need ot rename the file as well

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (adamdougal) [inline docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:36]: I'd argue this is a benefit. Because it's just configuration, we won't need to test it. Any code based elements can still be unit tested.

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (adamdougal) [inline docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:42]: Could we put the good at the top of the list? My OCD is triggering a bit here 🙈

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (adamdougal) [inline docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:35]: This is the same for langchain vs openai functions vs semantic kernel. Do you think we should align on just one orchestrator as a favourite?

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (adamdougal) [inline docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:34]: I might be being optimistic, but I don't think it will be significant effort. Is it work breaking it down and doing rough estimates?

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (superhindupur) [reply docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:35]: Isn't semantic kernel a code-based orchestrator? If that's the case wouldn't the interfaces in code ensure that the functionality will remain common across it and other code-based orchestrators? If we don't have interfaces already we at least have the possibility of creating them and enforcing uniformity if they're in code.

With PF, that's not an option because the orchestration is in a dag file.

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (superhindupur) [reply docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:36]: > Because it's just configuration, we won't need to test it

e2e behavior will need to be tested for correctness, ordering of actions etc.

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (superhindupur) [reply docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:34]: Fair enough, the "significant effort" is subjective. My suggestion is - let's hold off on this until we're able to validate customer requirements with Randy tomorrow and maybe we will have a strong argument in favor of adding promptflow.

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (adamdougal) [reply docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:36]: Yea, but the negative of there being no way to test it, isn't a negative if you don't need to test it.

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (adamdougal) [reply docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:35]: > Isn't semantic kernel a code-based orchestrator?

Yea correct. There's only one common method which is:

```python
    @abstractmethod
    def orchestrate(
        self, user_message: str, chat_history: List[dict], **kwargs: dict
    ) -> dict:
        pass
```
https://github.com/Azure-Samples/chat-with-your-data-solution-accelerator/blob/3e71fe10c50a129a20c314ba015065acdc6d8379/code/backend/batch/utilities/orchestrator/OrchestratorBase.py#L27-L31

Each orchestrator is then free to do what it likes within that. 

Obviously it's extra effort, but we should ensure as reviewers all changes made and features added are applied to all orchestration types.

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (ross-p-smith) [reply docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:1]: How about `2024-04-23-prompflow-orchestration.md`

Comment (superhindupur) [reply docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:1]: Not applicable anymore since the decision matches with the file name.

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (superhindupur) [reply docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:37]: Fair enough, took this into consideration while making the final decision.

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (superhindupur) [reply docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:38]: I disagree with the point that we don't need to test it.

BUT the lack of e2e testing applies to all orchestrators at the moment. And when e2e tests are added, they can quite easily cover the promptflow backend too. Incorporated this into the final decision.

File: docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md
Comment (adamdougal) [inline docs/design/adrs/2024-04-23-implement-prompflow-orchestration.md:3]:
```suggestion
* **Status:** approved
```

**Review by adamdougal — APPROVED** [review]

**Review by adamdougal — APPROVED** [review]

**Pull request merged**
